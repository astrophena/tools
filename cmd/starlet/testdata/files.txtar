-- bot.star --
def handle(update):
    # Should succeed.
    json = files.read("update.json")
    # Should return an error.
    nonexistent = files.read("nonexistent.md")
-- update.json --
{
  "update_id": 10000,
  "message": {
    "date": 1441645532,
    "chat": {
      "last_name": "Test Lastname",
      "id": 123456789,
      "first_name": "Test",
      "username": "Test"
    },
    "message_id": 1365,
    "from": {
      "last_name": "Test Lastname",
      "id": 123456789,
      "first_name": "Test",
      "username": "Test"
    },
    "text": "/start"
  }
}
