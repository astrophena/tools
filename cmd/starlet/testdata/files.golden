[
  {
    "method": "sendMessage",
    "args": {
      "chat_id": 123456789,
      "entities": [
        {
          "length": 120,
          "offset": 24,
          "type": "pre"
        }
      ],
      "link_preview_options": {
        "is_disabled": true
      },
      "text": "‚ùå Something went wrong:\nTraceback (most recent call last):\n  bot.star:5:29: in handle\nError in files.read: file nonexistent.md not found in Gist\n"
    }
  }
]